### PR TITLE
Add API endpoint for fetching continuations

### PR DESCRIPTION
**Pull request by KentShikama**
## Overview
Provide a brief description of what this PR does, and why it's needed.

Continuation of #1274 

The GRPC endpoint is defined as follows

```
rpc listenForContinuationAtName(Channels) returns (ListeningNameContinuationResponse) {}

message Channels {
  repeated Channel channels = 1;
}

message ListeningNameContinuationResponse {
  string status = 1;
  repeated ContinuationsWithBlockInfo blockResults = 2;
  int64 length = 3;
}

message ContinuationsWithBlockInfo {
  repeated WaitingContinuationInfo postBlockContinuations = 1;
  BlockInfoWithoutTuplespace block = 2;
}

message WaitingContinuationInfo {
  repeated BindPattern postBlockPatterns = 1;
  Par postBlockContinuation = 2;
}

message BlockInfoWithoutTuplespace {
  string blockHash = 1;
  string blockSize = 2;
  int64 blockNumber = 3;
  int64 version = 4;
  int32 deployCount = 5;
  string tupleSpaceHash = 6; // Same as postStateHash of BlockMessage
  int64 timestamp = 7;
  float faultTolerance = 8;
  string mainParentHash = 9;
  repeated string parentsHashList = 10;
  string sender = 11;
}

```

### Which JIRA issue does this PR relate to? If there is not a JIRA issue addressing this work, please create one now and add the link here.

https://rchain.atlassian.net/browse/RHOL-441

Add link to corresponding JIRA issue.

### Complete this checklist before you submit the PR
- [X] This PR contains no more than 200 lines of code, excluding test code.
- [X] This PR meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards).
- [X] You have someone in mind to assign for review. Make this assignment after submitting the PR.

### Notes
Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else.

### CODE REVIEW

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala [hunk 1]
```diff
@@ -68,56 +68,120 @@ object BlockAPI {
...
+    }
+
+  private def isListeningNameReduced(
+      block: BlockMessage,
+      sortedListeningName: immutable.Seq[Channel])(implicit channelCodec: Codec[Channel]) = {
+    val serializedLog =
+      block.body.fold(Seq.empty[Event])(_.commReductions)
+    val log =
+      serializedLog.map(EventConverter.toRspaceEvent).toList
+    log.exists {
```
Comment (birchmd) [inline casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: Is this enough? For example, if either the produce or consume is already in the tuplespace when the other half arrives to make a comm event then we still want to report that right? So do we need to look for the `COMM` case as well?

**Comment by KentShikama** [comment]
That is for https://rchain.atlassian.net/browse/RHOL-599

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala [hunk 2]
```diff
@@ -127,6 +125,32 @@ class RuntimeManager private (val emptyStateHash: ByteString, runtimeContainer:
...
+  }
+
+  def getContinuation(hash: ByteString,
+                      channels: immutable.Seq[Channel]): Seq[(Seq[BindPattern], Par)] = {
+    val resetRuntime = getResetRuntime(hash)
+    val results: Seq[WaitingContinuation[BindPattern, TaggedContinuation]] =
+      resetRuntime.space.getWaitingContinuations(channels)
+    runtimeContainer.put(resetRuntime)
+    if (results.exists(_.continuation.taggedCont.isScalaBodyRef)) {
+      throw new Error("The results contain a primitive/native scala function.")
```
[the remark this answers is not in the document]
Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: Silently filter them out?

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala (hunk 1 above)
Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: Great catch; will do

File: casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala [hunk 3]
```diff
@@ -68,56 +68,120 @@ object BlockAPI {
...
+        maybeBlocksWithActiveName <- mainChain.toList.traverse { block =>
+                                      getContinuationsWithBlockInfo[F](runtimeManager,
+                                                                       sortedListeningNames,
+                                                                       block)
+                                    }
+        blocksWithActiveName = maybeBlocksWithActiveName.flatten
+      } yield
+        ListeningNameContinuationResponse(status = "Success",
+                                          blockResults = blocksWithActiveName,
+                                          length = blocksWithActiveName.length.toLong)
```
[the remark this answers is not in the document]
Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala]: Sounds good

**Comment by KentShikama** [comment]
@birchmd PTAL

File: casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (KentShikama) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/RuntimeManager.scala]: I guess it is fine

**Pull request merged**